components/MashTags/MashTags.tsx

```diff
@@ -0,0 +1,111 @@
+import moment from 'moment';
+import { MashReferral, ReferralStage } from 'types';
+
+interface Props {
+  mashReferral: MashReferral;
+}
+const highRating = 4 * 60 * 60 * 1000;
+const mediumRating = 24 * 60 * 60 * 1000;
+const lowRating = 72 * 60 * 60 * 1000;
+
+const initialLowRatings = new Set([
+  'E3 REFERRAL',
+  'EH SCREENING REQUIRED IN MASH',
+]);
+
+const initialMedRatings = new Set(['CSC SCREENING REQUIRED IN MASH']);
+
+const initialHighRatings = new Set(['PROGRESS STRAIGHT TO CSC ALLOCATION']);
+
+const output = (mashReferral: MashReferral) => {
+  const currentTime = new Date().toISOString();
+  const hoursPassed = moment(currentTime, 'YYYY-MM-DD[T]HH:mm:ss. SSS[Z]').diff(
+    moment(mashReferral.createdAt, 'YYYY-MM-DD[T]HH:mm:ss. SSS[Z]'),
+    'hours'
+  );
+  const minsPassed = moment(currentTime, 'YYYY-MM-DD[T]HH:mm:ss. SSS[Z]').diff(
+    moment(mashReferral.createdAt, 'YYYY-MM-DD[T]HH:mm:ss. SSS[Z]'),
+    'minutes'
+  );
+  let timeLeftinMilliseconds = 0;
+
+  if (
+    mashReferral.stage === ReferralStage.SCREENING ||
+    mashReferral.stage === ReferralStage.FINAL
+  ) {
+    const initialhoursPassed = moment(
+      currentTime,
+      'YYYY-MM-DD[T]HH:mm:ss. SSS[Z]'
+    ).diff(
+      moment(mashReferral.initialCreatedAt, 'YYYY-MM-DD[T]HH:mm:ss. SSS[Z]'),
+      'hours'
+    );
+
+    if (
+      initialLowRatings.has(mashReferral.initialDecision?.toUpperCase() || '')
+    ) {
+      timeLeftinMilliseconds = lowRating - initialhoursPassed * 60 * 60 * 1000;
+    }
+    if (
+      initialMedRatings.has(mashReferral.initialDecision?.toUpperCase() || '')
+    ) {
+      timeLeftinMilliseconds =
+        mediumRating - initialhoursPassed * 60 * 60 * 1000;
+    }
+    if (
+      initialHighRatings.has(mashReferral.initialDecision?.toUpperCase() || '')
+    ) {
+      timeLeftinMilliseconds = highRating - initialhoursPassed * 60 * 60 * 1000;
+    }
+  } else if (mashReferral.stage === ReferralStage.INITIAL) {
+    timeLeftinMilliseconds = mediumRating - hoursPassed * 60 * 60 * 1000;
+  }
+  const isOverdue = timeLeftinMilliseconds < 0;
+  return {
+    hoursPassed,
+    minsPassed,
+    timeLeftinMilliseconds,
+    isOverdue,
+  };
+};
+
+const MashTags = ({ mashReferral }: Props): React.ReactElement => {
+  const { hoursPassed, minsPassed, timeLeftinMilliseconds, isOverdue } =
+    output(mashReferral);
+
+  if (mashReferral.stage === ReferralStage.CONTACT) {
+    if (minsPassed < 60)
+      return (
+        <div className="govuk-tag lbh-tag lbh-tag--green">
+          {minsPassed} mins ago
+        </div>
+      );
+    if (hoursPassed > 1 || hoursPassed == 1)
```

Comment:
watch out for ==

Also this could just an else block, if minsPassed is not less than 60 then an hour or more must have passed, no need for the conditional